syntax = "proto3";

package kvstore;

service KeyValueOps
{
    rpc GetFromDB(GetRequest) returns (GetReply) {}
    rpc PutToDB(PutRequest) returns (PutReply) {}
}

message GetRequest {
    string key = 1;
    uint32 consistency_level = 2;
}

message GetReply {
    string value = 1;
}

message PutRequest {
    string key = 1;
    string value = 2;
}

message PutReply {
}